GVFS/GVFS.Common/Git/GitAuthentication.cs

```diff
@@ -37,57 +38,62 @@ public bool IsBackingOff
 
         public bool IsAnonymous { get; private set; } = true;
 
-        public void ConfirmCredentialsWorked(string usedCredential)
+        public void ApproveCredentials(string credentialString)
         {
             lock (this.gitAuthLock)
             {
-                if (usedCredential == this.cachedAuthString)
+                // Don't reset the backoff if this is for a different credential than we have cached
+                if (credentialString == this.cachedCredentialString)
                 {
                     this.numberOfAttempts = 0;
                     this.lastAuthAttempt = DateTime.MinValue;
+
+                    // Tell Git to store the valid credential if we haven't already
+                    // done so for this cached credential.
+                    string username;
+                    string password;
+                    if (!this.isCachedCredentialStringApproved &&
+                        TryParseCachedAuthString(this.cachedCredentialString, out username, out password))
+                    {
+                        this.git.ApproveCredentials(this.repoUrl, username, password);
+                        this.isCachedCredentialStringApproved = true;
+                    }
                 }
             }
         }
 
-        public void Revoke(string usedCredential)
+        public void RejectCredentials(string credentialString)
         {
             lock (this.gitAuthLock)
-            {
-                if (usedCredential != this.cachedAuthString)
-                {
-                    // Don't stomp a different credential
-                    return;

```

Comment:
I inverted the `if (..x..) return; DoThing();` to `if (!..x..) { DoThing(); }` to be consistent with the way we do this check in the above `CredentialsWorked`/`Approve` method.